src/solver/simulation/economy.cpp

```diff
@@ -143,9 +142,11 @@ bool Economy::year(Progression::Task& progression,
         pProblemesHebdo[numSpace].weekInTheYear = state.weekInTheYear = w;
         pProblemesHebdo[numSpace].HeureDansLAnnee = hourInTheYear;
 
-        ::SIM_RenseignementProblemeHebdo(state.study,
-          pProblemesHebdo[numSpace], state.weekInTheYear, numSpace, hourInTheYear,
-          valeursGenereesParPays);
+        ::SIM_RenseignementProblemeHebdo(study, pProblemesHebdo[numSpace], state.weekInTheYear, 
```

Comment:
Note that we add a call to a new function **BuildThermalPartOfWeeklyProblem**. It's an extraction from **SIM_RenseignementProblemeHebdo** (which is gigantic). We first needed to extract it because of adequacy patch mode calling **SIM_RenseignementProblemeHebdo** a second time and erasing thermal production costs (no further details, sorry). We don't especially need this extraction anymore, but we chose to leave it here, as a remainder that we should split the gigantic **SIM_RenseignementProblemeHebdo** into several simpler ones.